pkg/controlplane/instance.go

```diff
@@ -499,5 +499,14 @@ func DefaultAPIResourceConfigSource() *serverstorage.ResourceConfig {
 	// enable the legacy beta resources that were present before stopped serving new beta APIs by default.
 	ret.EnableResources(legacyBetaEnabledByDefaultResources...)
 
+	// There are some features that require some APIs resources and versions to be enabled.
+	// In order to avoid users to break their clusters by enabling the feature gate but
+	// not enabling the corresponding API groups, we can conditionally enable these APIs for them.
+	if utilfeature.DefaultFeatureGate.Enabled(features.MultiCIDRServiceAllocator) {
```

Comment:
@liggitt please take a look at this approach, alternatives to enable the feature beta by default are:
- this approach
- ask users to enable the corresponding apis, but clusters by default will not be able to start until those are enabled
- graduate the API groups to GA, but that has a risk that the APIs does not have feedback of the users running the feature in production

Reply:
for previous betas like this:
1. we've left the gate off by default
2. required both the gate and the API to be enabled together, e.g. `--feature-gates=MultiCIDRServiceAllocator=true --runtime-config=networking/v1beta1/ipaddresses=true,networking/v1beta1/servicecidrs=true`
3. validate user doesn't enable the gate and not the API

Reply:
how do we get feedback with that approach during beta?
This feature really needs feedback with beta enabled by default

Reply:
> This feature really needs feedback with beta enabled by default

This is super tough, but if it depends on a beta endpoint that will not exist permanently once the feature graduates, we're not going to enable it by default

Reply:
make sense, I will drop this approach